### PR TITLE
[BISERVER-15067] Add toggle property to enable/disable admin access to other users trash

### PR DESCRIPTION
**Pull request by befc**
Added the property `adminAccessAllUsersTrash` to the `pentaho.xml` configuration file, by default it will keep the current behaviour of showing all user thrash to admin users, if disabled trash items of other users will not be shown to admins.

Should be merged together with: https://github.com/pentaho/pentaho-platform-ee/pull/1764

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 22m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-solutions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **458**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5636/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by renato-s — APPROVED** [review]

File: repository/src/test/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelperTest.java [hunk 1]
```diff
@@ -385,6 +388,17 @@ protected List<String> getUserList() {
...
+    // as admin all user trash access disabled
+    try ( MockedStatic<PentahoSystem> mockedPentahoSystem = mockStatic( PentahoSystem.class ) ) {
+      mockedPentahoSystem.when(
+        () -> PentahoSystem.getSystemSetting( "adminAccessAllUsersTrash", "false" ) )
+        .thenReturn( "false" );
+      admin[0] = true;
+      final List<RepositoryFile> deletedFilesAdminNoAccess = defaultDeleteHelper.getAllDeletedFiles( session, pentahoJcrConstants );
+      assertNotNull( deletedFilesAdminNoAccess );
+      assertEquals( 1, deletedFilesAdminNoAccess.size() );
+    }
```
Comment (smmribeiro) [inline repository/src/test/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelperTest.java:401]: Please check if it is possible to add a test for the scenario  would be great if there could be a test for when adminAccessAllUsersTrash is set to true;

**Review by smmribeiro — CHANGES_REQUESTED** [review]

File: repository/src/test/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelperTest.java (hunk 1 above)
Comment (befc) [reply repository/src/test/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelperTest.java]: Added, also refactored the unit tests to have one test method per test case, while reusing the setup instructions code

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 35s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
assemblies/pentaho-solutions,repository
```
---
:ok_hand: All tests passed!
 
Tests run: **464**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5636/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by smmribeiro — APPROVED** [review]
Looks good.

**Pull request merged**